### ISSUE TITLE
Legibility of documentation

### ISSUE THREAD

**Issue by LocalStarlight** [ebruneton/precomputed_atmospheric_scattering#19]
My apologies if this is a really stupid question.

I am trying to read this code in detail so that I can really understand it, but the formatting of the documentation is very strange.

It kinda looks like HTML (with all the paragraph tags etc. and SVG sections) but when I copy-pasted the code into an HTML file and opened it up, it's all over the place.

And where on an SVG diagram a point is labelled, for example, 'b', it's listed in the documenation like this: $\bp$

And then there are sections like this: $\mu=\bo\bp\cdot\bp\bi/\Vert\bo\bp\Vert\Vert\bp\bi\Vert$

Which I presume should look a lot simpler!

How can I view this code in the way intended so that it is intelligible?

Thanks!

**Comment by ebruneton** [comment]
You can use "make doc" to generate the documentation in HTML format. The result should look like https://ebruneton.github.io/precomputed_atmospheric_scattering/

**Comment by LocalStarlight** [comment]
Hey, sorry to be annoying! I thought maybe at first you were referring to some feature in Github itself, but I can't find any way to "make doc". Then I thought maybe you were referring to this program: https://github.com/1602/makedoc

But I tried installing that and running it, for example:

makedoc functions.glsl

And it just throws an error:

`/Users/XXXX/Documents/GitHub/precomputed_atmospheric_scattering/atmosphere/functions.glsl 0 0 1
/usr/local/lib/node_modules/makedoc/lib/documentation.js:45
    if (path.existsSync(dir)) {
             ^

TypeError: path.existsSync is not a function
    at Documentation.setOutputDir (/usr/local/lib/node_modules/makedoc/lib/documentation.js:45:14)
    at Project.makeDocumentation (/usr/local/lib/node_modules/makedoc/lib/project.js:111:9)
    at done (/usr/local/lib/node_modules/makedoc/lib/project.js:81:48)
    at /usr/local/lib/node_modules/makedoc/lib/project.js:61:13
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:511:3)`

Not sure what I'm doing wrong, sorry!

**Comment by LocalStarlight** [comment]
Hey, I'm going a bit mad trying to figure this out.

Can you be a bit more specific about how to create the documentation? (Sorry!)

**Comment by ebruneton** [comment]
PS: I'm referring to https://en.wikipedia.org/wiki/Make_(software)#Modern_versions (more precisely GNU Make), to be used with https://github.com/ebruneton/precomputed_atmospheric_scattering/blob/master/Makefile

**Comment by LocalStarlight** [comment]
Hey, thanks for following up. I actually figured out that the full documentation (https://ebruneton.github.io/precomputed_atmospheric_scattering/) has links to all the documentation files I was looking for anyway. Thanks so much for your help.

And thanks for this incredible work. I have a mountain to climb trying to understand it (haven't done any maths since I was in school), but hoping to get it working in Unreal Engine.

**Comment by ebruneton** [comment]
By the way, the default atmosphere in Unreal Engine is already using this algorithm* (more precisely the previous implementation, but it produces similar results as this new implementation): https://github.com/sinisterrook/unrealengine/blob/master/Engine/Shaders/AtmosphereCommon.usf.

*maybe with some simplifications or modifications, I haven't checked their code.

**Comment by LocalStarlight** [comment]
Ah, I did not know that!

Unless I'm mistaken, it doesn't seem to be set up for a planetary atmosphere though? I need to be able to enter/exit the atmosphere from space, it seems like UE4's version is based around a flat plane, but I could be wrong. I've spent the last week or so trying to understand and figure out your code (not ever having used glsl before has slowed me down considerably), so hopefully that wasn't all in vain if I could have just used UE4's own system anyway! Thanks for letting me know.
